src/mds/MDSRank.cc

```diff
@@ -1930,6 +1930,17 @@ void MDSRank::resolve_done()
   snapclient->sync(new C_MDSInternalNoop);
 }
 
+static size_t apply_blocklist(MDSRank *mds, Server *server,
```

Comment:
Having random static methods is just weird, at least within the Ceph codebase. This should be a protected helper method of `MDSRank`.

Reply:
There is less reason to keep this as part of the class since all callers are in MDSRank source. But I do agree it is a bit off-putting...